DSA with C++/Singly_Linked_List.cpp

```diff
@@ -0,0 +1,88 @@
+// C++ program to implement singly linked list using a class
```

Comment:
Strengths:

1. The code is clear and well-structured, demonstrating good use of classes for the Node and LinkedList.
2. Functionality like insertAtHead and print are straightforward and easy to follow.
3. The parameterized constructor for Node is efficient and eliminates redundancy.
Suggestions:

1. The print function could use a cleaner traversal loop for simplicity, like a for loop.
2. Consider adding functionality to handle more operations like deletion or size computation.
3. Minor optimizations could be made to shorten the implementation without losing readability.
Overall, great foundational work!